Обнова интегральных плат, номер 2.

Doster-d opened the pull request:
Или "Интегральное".
Это продолжение идеи #709.
Очень нужна помощь в тестирование этого ПРа, физически и идейно не могу проверить все новые интегралки.

[Тут будет описание всех фич, которые появится после этого обновление]
- Были добавлены новые службы функции и дефайны для улучшения кода и жизни игроков Хаотичного Оникса/ЕОС.
- Проведен рефакторинг всего кода интегральных схем, этим можно было и закончить, но мы продолжим:
- Переделан способ передачи доступа для управления дверьми и прочими механизмами, теперь там служит она плата и багов с ней стало меньше.
- Переделана работа локатора консоли телепорт, функционал не потерян.
- Добавлена плата просмотра состояния батареи.
- Добавлена плата выбора одной строки из заданных строк, помогает если надо будет выбирать часть тела для хирургии или оружия.
- Добавлена плата анализа рюкзака, теперь намного проще будет узнавать, че это там у того ассистента в рюкзаке.
- Добавлена плата объединения списков, как будто ими кто-то пользовался, мех.
- Исправлен баг, что в интегральные платы при засовывание предмета, предмет выпадает на пол.
- В оружейную плату добавлен выбор части телу в которое будет стрелять оружие, немного скорректирован принцип работы самой оружейной платы, увеличен размер платы.
- Пуллер теперь может пулить любой подвижный объект.
- Добавлены плата вставления объекта в рюкзак, очень круто, про БОХ не забыл.
- Добавлена плата, которая меняет описание Вашей интегралки.
- Добавлена плата, которая меняет цвет Вашей интегралки.
- Добавлена плата, которая меняет имя Вашей интегралки, очень полезно с текст-ту-спич платой.
- И мое любимое - добавлена плата, которая может проводить операцию в зависимости от того какой инструмент вставлен в нее.
- Добавлены шахтерские платы, которые анализируют, копают и собирают ресурсы с турфа в ящик.
- Изменена работа камеры, теперь она может транслировать в любой канал камер.
- Добавлены плата со звуками HEV'а из половина жизни 2, добавлены звуки системшока и соответственно плата
- Перенес работу со строками (текстом) в отдельный раздел.
- Добавлена плата разбрызгивателя жидкости, добавлена плата к которой можно прикрепить стакан.
- Плата "управления ИИ" было разделена на две части, на контроль через ММИ и контроль через пИИ, возможность вставить ИИ была удалена (требуется проверка работоспособности этих плат).
- Добавлена плата, которая активирует пин при перемещение платы
- Появился новый раздел питание (что хотелось добавить автору в #709), теперь Вы можете заряжать сеть/строить бюджетный powersink, в этот раздел хоть: повернет коннектор (powersink) и трансмиттеры энергии, которая заряжает батарейки, есть малая и большая версия.
- ~~Пофиксил~~ Убрал сохранение цвета интегралки.
- Добавлены новые платы, изменен функционал плат, которые не указаны выше, постараюсь дать описание чуть позже.
- Все интерфейсы были переведены на "фейкнаноуи", так же они были изменены для того, что бы было проще создавать интегралки.
- Добавлены новые корпуса, ничего про них сказать нельзя, корпуса, как корпуса.
- Добавлены расследовательские функции педалям, связанные с интегралками.

close #3971, close #4044, close #2316, close #2144, close #2143, close #2227, close #4249, close #3693

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и явных ошибок не было.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

Doster-d committed:
tweak(integrated_circuits) port from BeeStation - complete.

Я ебнулся.

TobyThorne commented:
![image](https://user-images.githubusercontent.com/45202681/103246148-afe1ec00-498c-11eb-84f7-f2fdf5ebcef0.png)
![image](https://user-images.githubusercontent.com/45202681/103246272-149d4680-498d-11eb-9348-908a784e5cd0.png)

Doster-d committed:
tweak(integrated_circuits) bug, style fixse

Немного обновил оружейную установку, теперь она классная.

Doster-d commented:
Чуть уменьшил, немного багов пофиксил, скоро будет очень круто.

Doster-d committed:
tweak(integrated_circuits) add new med and mining circuits

made some bug fixes

Doster-d committed:
Merge branch 'dev' into circ-upd

Doster-d committed:
plasma fix

Doster-d commented:
И года не прошло, а ПР уже готов к ре... Как это прошел?

Doster-d commented:
В этот раз атмосферных плат не будет.

Doster-d committed:
tweak(integrated_circuits) new circuit, some bug fixes

Doster-d commented:
Сделал список изменений, указал все, кроме мелких изменений в логике по типу более лучших проверок чисел.

Doster-d committed:
tweak(integrated_circuit) Move replaced to moved_event, bug fixes

Scadouge commented:
Pulling claw перестал работать вообще

Scadouge commented:
Не знаю было ли так задумано, но mining drill не может ломать huge rock и rocky debris.
![rock](https://user-images.githubusercontent.com/22453846/103532628-a846a980-4e9c-11eb-901e-f31f855be37b.png)

Doster-d commented:
> Не знаю было ли так задумано, но mining drill не может ломать huge rock и rocky debris.
> ![rock](https://user-images.githubusercontent.com/22453846/103532628-a846a980-4e9c-11eb-901e-f31f855be37b.png)

Ну естественно, он ломает только турфы с минералами, а надо ли что бы ломало их?

Scadouge commented:
Ну камней очень много, если делать что-то простое автономное, то они станут проблемой. На счет платы не знаю, но с ручного бура с тайла может выпасть rocky debris, который тоже заблокирует проход. Просто если обычный бур может их добывать/ломать, то почему бы и плате не дать такое?

Doster-d commented:
> Ну камней очень много, если делать что-то простое автономное, то они станут проблемой. На счет платы не знаю, но с ручного бура с тайла может выпасть rocky debris, который тоже заблокирует проход. Просто если обычный бур может их добывать/ломать, то почему бы и плате не дать такое?

Хорошо, сделаю, если есть какие-то вопросы или пожелания, то лучше обратитесь в личку дискорда ```Architector#4543```, тут могу не заметить.

Doster-d committed:
tweak(integrated circuits) new sounds ciruits, bug fixes

TobyThorne commented:
![image](https://user-images.githubusercontent.com/45202681/104108960-48645e80-52f3-11eb-8b68-3089acf2bce2.png)
Эти спрайты точно должны быть удалены? Почему должны быть удалены? Где упоминание в описании ПРа? Где квента на 1200 слов?

Doster-d committed:
tweak(integrated circuits) sprites go back

Doster-d commented:
> ![image](https://user-images.githubusercontent.com/45202681/104108960-48645e80-52f3-11eb-8b68-3089acf2bce2.png)
> Эти спрайты точно должны быть удалены? Почему должны быть удалены? Где упоминание в описании ПРа? Где квента на 1200 слов?

[Использует силу] Вы ничего не видели.

TobyThorne commented:
![image](https://user-images.githubusercontent.com/45202681/104109078-5f578080-52f4-11eb-89a3-5a7a54a8629a.png)
Эти спрайты точно должны быть удалены? Почему должны быть удалены? Где упоминание в описании ПРа? Где квента на 1200 слов?

Doster-d commented:
> ![image](https://user-images.githubusercontent.com/45202681/104109078-5f578080-52f4-11eb-89a3-5a7a54a8629a.png)
> Эти спрайты точно должны быть удалены? Почему должны быть удалены? Где упоминание в описании ПРа? Где квента на 1200 слов?

Первые две интегралки были удалены еще на ТГ, я с ними солидарен, не нужно.
Охлаждения/нагрев реагентов удален из билда давно, соответственно удалены и спрайты
Два спрайта для интегралок, которые были заменены действием с инструментами.
Интерактор никогда не был использован, соответственно удалено.

TobyThorne commented:
>Убрал сохранение цвета интегралки.

Чё это значит? Интегралки нельзя красить? Ничего не понятно.

Doster-d commented:
> > Убрал сохранение цвета интегралки.
> 
> Чё это значит? Интегралки нельзя красить? Ничего не понятно.

У нас есть такой инструмент, который называется анализатор, он позволяет сохранять код интегралки, раньше он сохранял цвет, теперь не сохраняет из-за боли в душе связанное с этим действием, красить их все еще можно, я бы написал, иначе если бы удалил изменения цвета интегралок.

TobyThorne commented:
Ладно, доверюсь описанию и воткну дизайнерский плюсик. 
А код - какой-то блять роман (судя по объему) со сранными платами. Не буду это читать.

Doster-d committed:
tweak(integrated ciruits) save circuits bug fix

Doster-d commented:
> А код - какой-то блять роман (судя по объему) со сранными платами. Не буду это читать.

Намного меньше, чем в #709.

TobyThorne reviewed (dismissed):
Бегло почитал, критического вроде бы не должно быть.
Если всё поломается и ничего не будет работать - слапайте Достера в дискорде. Прям наверняка. Шоб пылынькало без остановок.

Doster-d committed:
Merge branch 'dev' into circ-upd

Doster-d commented:
Не принимайте.

kreeperHLC commented:
> Не принимайте.

Внутривенно иль снаружи?

Doster-d commented:
> Внутривенно иль снаружи?

Интегрально

Doster-d committed:
tweak(integrated circuits) fixed mmi tank & pAI connector movement

Добавил хендлеры мобам и все заработало, зрада!

MACTEP-ABATAP commented:
Нужно написать бота, который выбранного человека будет без остановок слапать.

Doster-d committed:
tweak(integrated_circuit) minor bug fixes

HonkyDonky commented:
Тут изменений на 3500 строчек, а вы двумя ревьювами хотите сказать, что ничего кринжового не нашли? Rlly?

kreeperHLC commented:
> Тут изменений на 3500 строчек, а вы двумя ревьювами хотите сказать, что ничего кринжового не нашли? Rlly?

Так это же порт, проще просто посмотреть иссуи с билда, с которого покрали. Ну и глазами пробежался чтобы не было бекдоров

HonkyDonky commented:
> Так это же порт

Тебе ли не знать, что порт может быть сломанным? 

К тому же автор отсебятины добавил. И может даже бекдуры, которые вы пропустили.

Doster-d committed:
tweak(integrated circuits) feat comeback, bug fixes

- Исправил странную ошибку с двойным инитиалайзом, теперь он один.
- Добавил прок GetAccess, теперь с помощью интегралки можно открывать двери и делать прикольные вещи
- Починил новую интегральный компонент из серии "реагенты" "огнетушитель", теперь она работает.
- Починил "scanner", если там было указано, что PUT DOWN равен FALSE, то он все равно забирал себе предмет, теперь такое не наблюдается.
- Добавлено уведомление о сканирование карты в кард ридер, добавлена новая-старая интегралка, которая меняет допуск на интегралки при помощи passkey
- Исправил неправильную логику работы одного из хелпера, не существенно.

Doster-d commented:
Произошел сброс аппрувов вот, что изменилось:
- Исправил странную ошибку с двойным инитиалайзом, теперь он один.
- Добавил прок GetAccess, теперь с помощью интегралки можно открывать двери и делать прикольные вещи
- Починил новую интегральный компонент из серии "реагенты" "огнетушитель", теперь она работает.
- Починил "scanner", если там было указано, что PUT DOWN равен FALSE, то он все равно забирал себе предмет, теперь такое не наблюдается.
- Добавлено уведомление о сканирование карты в кард ридер, добавлена новая-старая интегралка, которая меняет допуск на интегралки при помощи passkey
- Исправил неправильную логику работы одного из хелпера, не существенно.

Doster-d commented:
Принимайте.

Doster-d commented:
конфликтный чел, депрессивный.

Doster-d committed:
Merge branch 'dev' into circ-upd

Doster-d commented:
не конфликтный чел, веселый (gay).

TobyThorne commented on code/modules/integrated_electronics/core/analyzer.dm:
```diff
@@ -5,14 +5,18 @@
-			to_chat(user, "<span class='notice'>You scan [A].</span>")
-			show_browser(user, saved, "window=circuit_scan;size=500x600;border=1;can_resize=1;can_close=1;can_minimize=1")
+			to_chat(user, SPAN("notice", "You scan [A]."))
+			user << browse(saved, "window=circuit_scan;size=500x600;border=1;can_resize=1;can_close=1;can_minimize=1")
```
почему

Doster-d replied:
Потому, исправил.

TobyThorne commented on code/modules/integrated_electronics/subtypes/memory.dm:
```diff
@@ -19,7 +19,7 @@
 
 /obj/item/integrated_circuit/memory/examine(mob/user)
-	. = ..()
+	..()
```
почему

Doster-d replied:
Моча в голову ударило, поэтому и изменил (так было на ТГ)

DeltaEpsilon7787 commented on code/modules/integrated_electronics/core/helpers.dm:
```diff
@@ -123,16 +120,52 @@
+		return
+	var/r
+	for(var/i = 1 to length(string))
+		r += ascii2text(text2ascii(string,i) ^ text2ascii(key,((i-1)%length(string))+1))
```
length(string) явно должен быть length(key), иначе смысла от этого никакого нет. Иначе строки длиной больше чем key будет вызывать выход за пределы строки.

Doster-d replied:
Ладно.

Doster-d replied:
Исправил.

Doster-d committed:
tweak(integrated_circuits) bug fixes

Doster-d commented:
- Увеличил лимиты у платы соединения текста в 2 раза
- Пофиксил фейковое сообщение об пожирания стали принтером
- Граббер теперь забирает  предмет, только если он на земле (предыдущие проверки сохранены).

Doster-d commented:
1 есть, остался еще 1, таким темпами я не смогу закончить свой смежный проект (шутка)

TobyThorne reviewed (approved):
каемся и молимся

TobyThorne commented:
ой пиздец
